### PR TITLE
init home work branch

### PR DESCRIPTION
**Pull request by MaksLeseu**

### CODE REVIEW

File: src/s1-main/App.tsx [hunk 1]
```diff
@@ -9,6 +9,9 @@ function App() {
             <HW1/>
             {/*раскомментировать по ходу выполнения*/}
             <HW2/>
+
```
Comment (Alekseyfgl) [inline src/s1-main/App.tsx:12]: You need to do it.

Comment (Alekseyfgl) [reply src/s1-main/App.tsx:12]: fdsfsd

Comment (Alekseyfgl) [reply src/s1-main/App.tsx:12]: dgdfgdf

Comment (MaksLeseu) [reply src/s1-main/App.tsx:12]: Okaj

File: src/s1-main/App.tsx [hunk 2]
```diff
@@ -9,6 +9,9 @@ function App() {
             <HW1/>
             {/*раскомментировать по ходу выполнения*/}
```
Comment (Alekseyfgl) [inline src/s1-main/App.tsx:10]: good!!!

**Review by Alekseyfgl — COMMENTED** [review]
I checked all

File: src/s1-main/App.tsx (hunk 1 above)
Comment (MaksLeseu) [reply src/s1-main/App.tsx:12]: Okaj

File: src/s2-homeworks/hw02/HW2.tsx [hunk 3]
```diff
@@ -35,23 +36,27 @@ const defaultAffairs: any = [ // need to fix any
 ]
 
 // pure helper functions
-export const filterAffairs = (affairs: any, filter: any): any => { // need to fix any
+export const filterAffairs = (affairs: Array<AffairType>, filter: string): any => { // need to fix any
```
Comment (Alekseyfgl) [inline src/s2-homeworks/hw02/HW2.tsx:39]: fix any from return

File: src/s2-homeworks/hw02/HW2.tsx [hunk 4]
```diff
@@ -35,23 +36,27 @@ const defaultAffairs: any = [ // need to fix any
...
 
+    if (filter === 'all') return affairs;
+    if (filter === 'low') return affairs.filter(i => i.priority === 'low')
+    if (filter === 'high') return affairs.filter(i => i.priority === 'high');
+    if (filter === 'middle') return affairs.filter(i => i.priority === 'middle');
 
-    return affairs // need to fix
 }
-export const deleteAffair = (affairs: any, _id: any): any => { // need to fix any
+export const deleteAffair = (affairs: Array<AffairType>, _id: number): any => { // need to fix any
```
Comment (Alekseyfgl) [inline src/s2-homeworks/hw02/HW2.tsx:47]: fix any from return

File: src/s2-homeworks/hw02/HW2.tsx [hunk 5]
```diff
@@ -35,23 +36,27 @@ const defaultAffairs: any = [ // need to fix any
...
 
-    return affairs // need to fix
+   return affairs.filter(i => i._id !== _id);
 }
 
 function HW2() {
     const [affairs, setAffairs] = useState<any>(defaultAffairs) // need to fix any
```
Comment (Alekseyfgl) [inline src/s2-homeworks/hw02/HW2.tsx:53]: fix any from useState

File: src/s2-homeworks/hw02/affairs/Affairs.tsx [hunk 6]
```diff
@@ -11,17 +11,17 @@ type AffairsPropsType = {
 }
 
 function Affairs(props: AffairsPropsType) {
-    const setAll = () => {
-        // need to fix
+    const setAll = (item: any) => {
```
Comment (Alekseyfgl) [inline src/s2-homeworks/hw02/affairs/Affairs.tsx:14]: What does item mean in params? I don't understand. 
and fix any

File: src/s2-homeworks/hw02/affairs/Affairs.tsx [hunk 7]
```diff
@@ -11,17 +11,17 @@ type AffairsPropsType = {
...
     }
-    const setMiddle = () => {
-        // need to fix
+    const setMiddle = (item: any) => {
+        props.setFilter(item);
     }
-    const setLow = () => {
-        // need to fix
+    const setLow = (item: any) => {
+        props.setFilter(item);
```
Comment (Alekseyfgl) [inline src/s2-homeworks/hw02/affairs/Affairs.tsx:24]: all the same

File: src/s2-homeworks/hw02/affairs/affair/Affair.tsx [hunk 8]
```diff
@@ -10,8 +10,8 @@ type AffairPropsType = {
 }
 
 function Affair(props: AffairPropsType) {
-    const deleteCallback = () => {
-        // need to fix
+    const deleteCallback = (id: number) => {
+        return props.deleteAffairCallback(id);
```
Comment (Alekseyfgl) [inline src/s2-homeworks/hw02/affairs/affair/Affair.tsx:14]: remove return

File: src/s2-homeworks/hw02/affairs/affair/Affair.tsx [hunk 9]
```diff
@@ -35,6 +35,7 @@ function Affair(props: AffairPropsType) {
             <button
                 id={'hw2-button-delete-' + props.affair._id}
                 className={buttonClass}
+                onClick={ () => deleteCallback(props.affair._id)}
                 // need to fix
```
Comment (Alekseyfgl) [inline src/s2-homeworks/hw02/affairs/affair/Affair.tsx:39]: add symbol into button

**Review by Alekseyfgl — CHANGES_REQUESTED** [review]

File: src/s2-homeworks/hw02/affairs/Affairs.tsx (hunk 6 above)
Comment (MaksLeseu) [reply src/s2-homeworks/hw02/affairs/Affairs.tsx:14]: I just fix any?

File: src/s2-homeworks/hw02/affairs/affair/Affair.tsx (hunk 9 above)
Comment (MaksLeseu) [reply src/s2-homeworks/hw02/affairs/affair/Affair.tsx:39]: What symbol?

File: src/s2-homeworks/hw02/affairs/Affairs.tsx [hunk 10]
```diff
@@ -11,17 +11,17 @@ type AffairsPropsType = {
 }
 
 function Affairs(props: AffairsPropsType) {
-    const setAll = () => {
-        // need to fix
+    const setAll = (item: string) => {
```
Comment (Alekseyfgl) [inline src/s2-homeworks/hw02/affairs/Affairs.tsx:14]: add void for returns

File: src/s2-homeworks/hw02/affairs/Affairs.tsx [hunk 11]
```diff
@@ -11,17 +11,17 @@ type AffairsPropsType = {
...
+        props.setFilter(item);
     }
-    const setHigh = () => {
-        // need to fix
+    const setHigh = (item: string) => {
```
Comment (Alekseyfgl) [inline src/s2-homeworks/hw02/affairs/Affairs.tsx:17]: add void for returns

File: src/s2-homeworks/hw02/affairs/affair/Affair.tsx [hunk 12]
```diff
@@ -10,8 +10,8 @@ type AffairPropsType = {
 }
 
 function Affair(props: AffairPropsType) {
-    const deleteCallback = () => {
-        // need to fix
+    const deleteCallback = (id: number) => {
```
Comment (Alekseyfgl) [inline src/s2-homeworks/hw02/affairs/affair/Affair.tsx:13]: add void for returns

**Review by Alekseyfgl — APPROVED** [review]

**Pull request merged**